### ISSUE TITLE
[Bug]: Links break when renaming a file

### ISSUE THREAD

**Issue by krolloc1** [carlonicora/obsidian-rpg-manager#324]
### Check for existing bug reports before submitting.

- [X] I searched for existing Bug Reports and found no similar reports.

### Expected Behavior

Rename a element and all elements linked to it should update their links

### Current behaviour

Rename an element, links get broken in other linked elements. The page then doesn't render.

### Reproduction

Rename an element, in the case and adventure element. Linked sessions don't render afterwards.

### Which Operating Systems are you using?

- [ ] Android
- [ ] iPhone/iPad
- [X] Linux
- [ ] macOS
- [ ] Windows

### Obsidian Version Check

1.4.16

### Plugin Version

4.1.12

### Confirmation

- [X] I have disabled all other plugins and the issue still persists.

**Comment by carlonicora** [comment]
Hi @krolloc1 

I have noticed the issue you are reporting and I have already dug deeper into the it.

## TL;DR
When obsidian sends a message that a file is updated, I update the links in the rest of the files, but they are being updated by obsidian themselves, so the two updates clash and the data loses some bits.

## Solution
Migrate from using the file path as identifier for the element to a uuid based.

## Challenges
Time

## Priority
High
